modules/packages/manifests/mozilla/generic_worker.pp

```diff
@@ -8,7 +8,11 @@
         'packages::mozilla::generic_worker::end': ;
     }
 
-    $tag = 'v10.8.4'
+    if ($tc_environment == 'staging') {
```

Comment:
I'm tempted to separate these (maybe we move it into the moco-config?) so that we can change staging in-repo instead of pinning to test. And then anyone could submit a pr to test a new version of generic-worker in the beta pool.

Reply:
`tc_environment` feels misleading as a name, especially when soon we'll have an actual staging taskcluster deployment. These are staging workers connected to the production taskcluster environment, rather than connected to a taskcluster staging environment. Perhaps `environment` is a better name than `tc_environment` since it is about the worker being a staging worker, rather than the taskcluster environment being a staging one. I realise you didn't invent the name, but if we're cleaning this up, this might be a good time to do it.

Reply:
:+1:  in releng build-puppet there are aspects for similar attributes, like security levels "high-security" etc and environments "prod" versus "dev". Using an aspect moves the problem to a naming concern for the value ("staging"), but then these workers can be perceived as releng-hardware "staging" and that avoids the conflict with "taskcluster staging".

Reply:
@petemoore how about this? https://github.com/davehouse/build-puppet/pull/1

Reply:
Thanks Dave, I'd say that looks perfect!

Reply:
@petemoore, the `tc_environment` name was encouraged by me because having a node def variable named `environment` was too generic for being in a top-level puppet namespace.  Puppet covers a lot of _other_ infrastructure.  I think @davehouse 's use of aspects is a great solution.

Reply:
@dividehex ah thanks for the clarification - that makes perfect sense - I hadn't realised that. ++